### PR TITLE
MueLu: Additional coarsening options

### PR DESCRIPTION
**Pull request by csiefer2**
Tests: RHEL7; New Test Added
Issue: N/A (non-ASC)

### CODE REVIEW

File: packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp [hunk 1]
```diff
@@ -171,34 +171,6 @@ namespace MueLu {
     FindNonZeros<Scalar,LocalOrdinal,GlobalOrdinal,Node>(myColsToZero->getData(0),dirichletCols);
   }
 
-
-  template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-  void ApplyRowSumCriterion(const Xpetra::Matrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>& A,
```
Comment (cgcgcg) [inline packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:176]: The Kokkos View version should be moved as well.

File: packages/muelu/doc/UsersGuide/masterList.xml [hunk 2]
```diff
@@ -411,6 +411,24 @@
...
+      <name>aggregation: classical scheme</name>
+      <type>string</type>
+      <default>"direct"</default>
+      <description>Prolongator formation option for classical coarsening.</description>
+      <visible>false</visible>
+      <comment-ML>parameter not existing in ML</comment-ML>
+    </parameter>
+
+    <parameter>
+      <name>aggregation: row sum drop tol</name>
```
Comment (cgcgcg) [inline packages/muelu/doc/UsersGuide/masterList.xml:424]: Remove `refmaxwell: row sum drop tol`.

File: packages/muelu/doc/UsersGuide/masterList.xml [hunk 3]
```diff
@@ -592,7 +610,7 @@
     <parameter>
       <name>aggregation: coloring algorithm</name>
       <type>string</type>
-      <default>serial</default>
+      <default>Zoltan2</default>
```
Comment (cgcgcg) [inline packages/muelu/doc/UsersGuide/masterList.xml:613]: Could this lead to issues? Serial is supposedly deterministic.

File: packages/muelu/src/Graph/Containers/MueLu_Zoltan2GraphAdapter.hpp [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+
+  Z2_STATIC_ASSERT_TYPES // validate the types
+};
+}//end namespace Zoltan2
+
+
+namespace MueLu {
+
+template <typename User, typename UserCoord=User>
+class MueLuGraphBaseAdapter : public Zoltan2::GraphAdapter<User,UserCoord> {
```
Comment (cgcgcg) [inline packages/muelu/src/Graph/Containers/MueLu_Zoltan2GraphAdapter.hpp:90]: Without checking any details: how is this different from the Zoltan2 Xpetra adapter?

File: packages/tpetra/core/src/Tpetra_CrsMatrix_decl.hpp [hunk 5]
```diff
@@ -1813,6 +1813,14 @@ namespace Tpetra {
                   Teuchos::ArrayRCP<const LocalOrdinal>& columnIndices,
                   Teuchos::ArrayRCP<const Scalar>& values) const;
 
+    /// Gets just the values array.  This *will* be a shallow copy
+    /// of the array (at least on the host memory space.  This
+    /// is not a const function, since the user can change these
+    /// values.
+    ///
+    /// \param values [out] Array of values. 
+    void getAllValues(Teuchos::ArrayRCP<Scalar>& values);
```
Comment (cgcgcg) [inline packages/tpetra/core/src/Tpetra_CrsMatrix_decl.hpp:1822]: Why is this needed? Couldn't you just use `getAllValues` and ignore rowptr and column indices?

**Review by cgcgcg — CHANGES_REQUESTED** [review]
That's a lot of code :-/

**Comment by jhux2** [comment]
This PR is a poster child for squashing commits.  🙏

File: packages/muelu/doc/UsersGuide/masterList.xml (hunk 3 above)
Comment (csiefer2) [reply packages/muelu/doc/UsersGuide/masterList.xml:613]: Yeah, but Zoltan2 works in parallel.

File: packages/muelu/src/Graph/Containers/MueLu_Zoltan2GraphAdapter.hpp (hunk 4 above)
Comment (csiefer2) [reply packages/muelu/src/Graph/Containers/MueLu_Zoltan2GraphAdapter.hpp:90]: That requires an Xpetra:CrsGrap under the hood, which GrapBase is *not*

File: packages/tpetra/core/src/Tpetra_CrsMatrix_decl.hpp (hunk 5 above)
Comment (csiefer2) [reply packages/tpetra/core/src/Tpetra_CrsMatrix_decl.hpp:1822]: In the Epetra case that actually copies the rowptr (because int vs. size_t)

File: packages/muelu/doc/UsersGuide/masterList.xml (hunk 2 above)
Comment (csiefer2) [reply packages/muelu/doc/UsersGuide/masterList.xml:424]: ?

Comment (csiefer2) [reply packages/muelu/doc/UsersGuide/masterList.xml:424]: I think we still need those guys for stuff at the top level.

File: packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp (hunk 1 above)
Comment (csiefer2) [reply packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:176]: Done

**Review by cgcgcg — APPROVED** [review]
